From 950ab37b019a9daab0c3f469b4d6c9d9aedfbf50 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 23 Jan 2025 11:29:14 +0100
Subject: [PATCH 1536/1695] dt-bindings: fix LTDC layer numbers on stm32mp2x
 platforms

On stm32mp2x platforms, the LTDC layers are name L1/2/3, not L0/1/2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git edd1aeeb1b562e81330e2fb4b182d8bca9a2a557 dt-bindings: fix LTDC layer numbers on stm32mp2x platforms]
Change-Id: If6d67d5f1b6c6ac2c51a00b11732e5428605ea5d
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/430960
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp215.dtsi                    | 4 ++--
 core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi  | 4 ++--
 core/arch/arm/dts/stm32mp231.dtsi                    | 4 ++--
 core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi  | 4 ++--
 core/arch/arm/dts/stm32mp251.dtsi                    | 4 ++--
 core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi  | 4 ++--
 core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi | 4 ++--
 core/include/dt-bindings/soc/stm32mp21-rifsc.h       | 4 ++--
 core/include/dt-bindings/soc/stm32mp25-rifsc.h       | 4 ++--
 9 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp215.dtsi b/core/arch/arm/dts/stm32mp215.dtsi
index 94e5dc2cd..3cb1082b0 100644
--- a/core/arch/arm/dts/stm32mp215.dtsi
+++ b/core/arch/arm/dts/stm32mp215.dtsi
@@ -19,11 +19,11 @@
 		access-controller-names = "cmn";
 
 		l1l2 {
-			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L0L1_ID>;
+			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L1L2_ID>;
 			access-controller-names = "l1l2";
 		};
 		l3 {
-			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L2_ID>;
+			access-controllers = <&rifsc STM32MP21_RIFSC_LTDC_L3_ID>;
 			access-controller-names = "l3";
 		};
 	};
diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
index b52406f10..84285dfef 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
@@ -72,8 +72,8 @@
 		RIFPROT(STM32MP21_RIFSC_SDMMC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_SDMMC3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_LTDC_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP21_RIFSC_LTDC_L0L1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP21_RIFSC_LTDC_L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP21_RIFSC_LTDC_L1L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP21_RIFSC_LTDC_L3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_CSI_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_DCMIPP_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_DCMI_PSSI_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 1fc045f30..397e0bd21 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -637,11 +637,11 @@
 				access-controller-names = "cmn";
 
 				l1l2 {
-					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L0L1_ID>;
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L1L2_ID>;
 					access-controller-names = "l1l2";
 				};
 				l3 {
-					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L2_ID>;
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L3_ID>;
 					access-controller-names = "l3";
 				};
 				rot {
diff --git a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
index 59591a6a1..4bc6ba9e9 100644
--- a/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp235f-dk-ca35tdcid-rif.dtsi
@@ -87,8 +87,8 @@
 		RIFPROT(STM32MP25_RIFSC_SDMMC3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_GPU_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L0L1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L1L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_ROT_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_TRIG_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index c12077835..df94faf3c 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -715,11 +715,11 @@
 				access-controller-names = "cmn";
 
 				l1l2 {
-					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L0L1_ID>;
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L1L2_ID>;
 					access-controller-names = "l1l2";
 				};
 				l3 {
-					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L2_ID>;
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L3_ID>;
 					access-controller-names = "l3";
 				};
 				rot {
diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
index 5a87d6ce1..f36d599b6 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
@@ -80,8 +80,8 @@
 		RIFPROT(STM32MP25_RIFSC_SDMMC3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_GPU_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L0L1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L1L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_ROT_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_TRIG_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
index 258f7c0d5..b5e9baa4f 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
@@ -80,8 +80,8 @@
 		RIFPROT(STM32MP25_RIFSC_SDMMC3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_GPU_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L0L1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP25_RIFSC_LTDC_L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L1L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
+		RIFPROT(STM32MP25_RIFSC_LTDC_L3_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_LTDC_ROT_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP25_RIFSC_DSI_TRIG_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
diff --git a/core/include/dt-bindings/soc/stm32mp21-rifsc.h b/core/include/dt-bindings/soc/stm32mp21-rifsc.h
index d7f394eea..6b896766f 100644
--- a/core/include/dt-bindings/soc/stm32mp21-rifsc.h
+++ b/core/include/dt-bindings/soc/stm32mp21-rifsc.h
@@ -91,8 +91,8 @@
 #define STM32MP21_RIFSC_I3C2_ID			115
 #define STM32MP21_RIFSC_I3C3_ID			116
 #define STM32MP21_RIFSC_ICACHE_DCACHE_ID	118
-#define STM32MP21_RIFSC_LTDC_L0L1_ID		119
-#define STM32MP21_RIFSC_LTDC_L2_ID		120
+#define STM32MP21_RIFSC_LTDC_L1L2_ID		119
+#define STM32MP21_RIFSC_LTDC_L3_ID		120
 #define STM32MP21_RIFSC_OTFDEC1_ID		125
 #define STM32MP21_RIFSC_IAC_ID			127
 
diff --git a/core/include/dt-bindings/soc/stm32mp25-rifsc.h b/core/include/dt-bindings/soc/stm32mp25-rifsc.h
index 2055bd0b9..19238c786 100644
--- a/core/include/dt-bindings/soc/stm32mp25-rifsc.h
+++ b/core/include/dt-bindings/soc/stm32mp25-rifsc.h
@@ -115,8 +115,8 @@
 #define STM32MP25_RIFSC_I3C3_ID			116
 #define STM32MP25_RIFSC_I3C4_ID			117
 #define STM32MP25_RIFSC_ICACHE_DCACHE_ID	118
-#define STM32MP25_RIFSC_LTDC_L0L1_ID		119
-#define STM32MP25_RIFSC_LTDC_L2_ID		120
+#define STM32MP25_RIFSC_LTDC_L1L2_ID		119
+#define STM32MP25_RIFSC_LTDC_L3_ID		120
 #define STM32MP25_RIFSC_LTDC_ROT_ID		121
 #define STM32MP25_RIFSC_DSI_TRIG_ID		122
 #define STM32MP25_RIFSC_DSI_RDFIFO_ID		123
-- 
2.47.2

